fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/NormalizedLuceneSearchProvider.java

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.dimension.impl;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionRow;
+import com.yahoo.bard.webservice.util.DimensionStoreKeyUtils;
+import com.yahoo.bard.webservice.util.Pagination;
+import com.yahoo.bard.webservice.web.util.PaginationParameters;
+import com.yahoo.bard.webservice.data.dimension.SearchQuerySearchProvider;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.miscellaneous.PerFieldAnalyzerWrapper;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.queryparser.simple.SimpleQueryParser;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Lucene search provider that also supports text search queries.
+ */
+public class NormalizedLuceneSearchProvider extends LuceneSearchProvider implements SearchQuerySearchProvider {
+    private static final Logger LOG = LoggerFactory.getLogger(NormalizedLuceneSearchProvider.class);
+
+    public static final String SEARCH_COLUMN_NAME = "__search";
```

Comment:
This looks like a pretty magical String. Is this something people need to know in order to build their
Lucene index to leverage this SearchProvider? If so, it should be documented in the class javadoc at least.

If it's just being used for something internally...well then it should be documented in the class javadoc
for future maintainers!